### PR TITLE
[Enhance](Nereids)add default order by of non-order query for nereids

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.7 seconds
 stream load tsv:          423 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230509091039_clickbench_pr_140833.html

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/NereidsAnalyzer.java [hunk 1]
```diff
@@ -57,6 +58,9 @@ public class NereidsAnalyzer extends BatchRewriteJob {
             bottomUp(
                 new CheckBound()
             ),
+            bottomUp(
+                    new AddDefaultOrderByForLimit()
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/NereidsAnalyzer.java:62]: should do it in rewriter before split limit and after merge filter and project

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/AddDefaultOrderByForLimit.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * add default order by for limit
+ * select * from t limit 10 offset 5
+ * to: select * from t order by {keys} limit 10 offset 5
+ */
+public class AddDefaultOrderByForLimit implements AnalysisRuleFactory {
+
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+                RuleType.ADD_DEFAULT_ORDER_BY_FOR_LIMIT.build(
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/AddDefaultOrderByForLimit.java:43]: should use custom rewrite to ensure limit is the root node

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/AddDefaultOrderByForLimit.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                        logicalLimit(logicalFilter(logicalOlapScan()))
+                                .then(limit -> limit.withChildren(ImmutableList.of(
+                                        new LogicalSort<>(buildOrderKeys(limit.child().child()), limit.child()))))),
+                RuleType.ADD_DEFAULT_ORDER_BY_FOR_LIMIT.build(
+                        logicalLimit(logicalOlapScan())
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/AddDefaultOrderByForLimit.java:48]: should also include limit(project(filter(scan)))

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**